packages/react/__tests__/get-label-from-stack-trace.js

```diff
@@ -0,0 +1,560 @@
+    expect(getLabelFromStackTrace(stackTrace)).toBe(expectedLabel)
+  })
+
+  test('Firefox', () => {
+    const stackTrace = `createEmotionProps@http://localhost:3000/static/js/bundle.js:46440:40
+jsx@http://localhost:3000/static/js/bundle.js:46636:113
+MyComponent$9@http://localhost:3000/static/js/bundle.js:47600:72
+renderWithHooks@http://localhost:3000/static/js/bundle.js:18904:27
+mountIndeterminateComponent@http://localhost:3000/static/js/bundle.js:21583:13
+beginWork@http://localhost:3000/static/js/bundle.js:22697:16`
+
+    expect(getLabelFromStackTrace(stackTrace)).toBe(expectedLabel)
+  })
+
+  test('Safari', () => {
+    const stackTrace = `createEmotionProps@http://localhost:3000/static/js/bundle.js:46440:49
+jsx@http://localhost:3000/static/js/bundle.js:46635:113
+renderWithHooks@http://localhost:3000/static/js/bundle.js:18904:27
+mountIndeterminateComponent@http://localhost:3000/static/js/bundle.js:21583:28
+beginWork$1@http://localhost:3000/static/js/bundle.js:27280:23`
+
+    expect(getLabelFromStackTrace(stackTrace)).toBeUndefined()
+  })
+})
+
+/**
+ * E.g.
+ *
+ * ```
+ * function MyComponent$9() {
+ *   return <div css={{ color: 'red' }} />
+ * }
+ * ```
+ */
+describe('typical function component', () => {
+  test('Chrome', () => {
+    const stackTrace = `Error
+    at createEmotionProps (emotion-element-895e3bbe.browser.esm.js:142)
+    at jsxDEV (emotion-react-jsx-dev-runtime.browser.esm.js:18)
+    at MyComponent$9 (App.js:5)
+    at renderWithHooks (react-dom.development.js:14803)
+    at mountIndeterminateComponent (react-dom.development.js:17482)
+    at beginWork (react-dom.development.js:18596)
+    at beginWork$1 (react-dom.development.js:23179)`
+
+    expect(getLabelFromStackTrace(stackTrace)).toBe(expectedLabel)
+  })
+
+  test('Firefox', () => {
+    const stackTrace = `createEmotionProps@http://localhost:3000/static/js/main.chunk.js:844:40
+jsxDEV@http://localhost:3000/static/js/main.chunk.js:1126:247
+MyComponent$9@http://localhost:3000/static/js/main.chunk.js:2274:92
+renderWithHooks@http://localhost:3000/static/js/vendors~main.chunk.js:16616:31
+mountIndeterminateComponent@http://localhost:3000/static/js/vendors~main.chunk.js:19228:17
+beginWork@http://localhost:3000/static/js/vendors~main.chunk.js:20306:20`
+
+    expect(getLabelFromStackTrace(stackTrace)).toBe(expectedLabel)
+  })
+
+  test('Safari', () => {
+    const stackTrace = `createEmotionProps@http://localhost:3000/static/js/main.chunk.js:844:49
+jsxDEV@http://localhost:3000/static/js/main.chunk.js:1126:247
+MyComponent$9@http://localhost:3000/static/js/main.chunk.js:2274:92
+renderWithHooks@http://localhost:3000/static/js/vendors~main.chunk.js:16616:31
+mountIndeterminateComponent@http://localhost:3000/static/js/vendors~main.chunk.js:19228:32
+beginWork$1@http://localhost:3000/static/js/vendors~main.chunk.js:24848:27`
+
+    expect(getLabelFromStackTrace(stackTrace)).toBe(expectedLabel)
+  })
+
+  test('SSR', () => {
+    const stackTrace = `Error
```

Comment:
huh, it's interesting that the same case has produced different-looking stack traces in Chrome and in node, even though they are both using V8 🤔

Reply:
It is interesting. It is likely because the Chrome stack trace is from CRA while the SSR one is from Next.js and there are differences in build tooling. Let me know if you want me to redo the Chrome one using the Next.js playground.